### PR TITLE
Practica 6: David Pérez Mijangos

### PR DESCRIPTION
**Pull request by DavidMijangos31**
Se agregan los archivos correspondientes a la Práctica 6.

### CODE REVIEW

File: README.md
Comment (umoqnier) [inline README.md:1]: Revierte este cambio, no queremos que elimines el README del repo

**Review by umoqnier — CHANGES_REQUESTED** [review]

File: practicas/David/p6/practica_6.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import json
+import torch
+from codecarbon import EmissionsTracker
+from datasets import DatasetDict
+
+tracker_training = EmissionsTracker()
+tracker_training.start()
```
Comment (umoqnier) [inline practicas/David/p6/practica_6.py:17]: Creo que se estila usar el tracker exclusivamente en la parte del entrenamiento, me pregunto que tanto afectará el registro de la carga de datos

File: practicas/David/p6/Readme.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   - La demo permite a los usuarios ingresar una reseña de película y el modelo devuelve la predicción del género (o géneros) correspondiente.
+
+
+## Requisitos
+- Python 3.11
+- Las bibliotecas mencionadas anteriormente deben estar instaladas.
+
+## Data
+Los datos utilizados para el entrenamiento del modelo de lenguaje provienen de Hugging Face: wykonos/movies
+- El dataset contiene reseñas de películas y sus respectivos géneros.
```
Comment (umoqnier) [inline practicas/David/p6/Readme.md:46]: Estaría chido ver los labels con los géneros por acá. Veo que los tienes en `labels.json` pero para tenerlo a la mano y mejorar la docu

File: practicas/David/p6/practica_6.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    "test": Dataset.from_pandas(test_df, preserve_index=False)
+})
+
+dataset
+
+model_name = "microsoft/deberta-v3-small"
+tokenizer = DebertaV2Tokenizer.from_pretrained(model_name)
+model = DebertaV2ForSequenceClassification.from_pretrained(
+    model_name,
+    num_labels=len(mapper),  # Ajusta al número de géneros
```
Comment (umoqnier) [inline practicas/David/p6/practica_6.py:50]: Hasta ahora pocos han hecho clasificación multi clase :D

File: practicas/David/p6/practica_6.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+def tokenize_function(examples):
+    return tokenizer(examples["text"], truncation=True, max_length=512, padding=False)
+
+tokenized_data = dataset.map(tokenize_function, batched=True)
+data_collator = DataCollatorWithPadding(tokenizer=tokenizer)
+
+cfl_metrics = evaluate.combine(["accuracy", "f1", "precision", "recall"])
+
+def sigmoid(x):
+    return 1 / (1 + np.exp(-x))
```
Comment (umoqnier) [inline practicas/David/p6/practica_6.py:70]: Me pregunto por qué elegiste esta función como función de activación O:

File: practicas/David/p6/practica_6.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+loss_history = pd.DataFrame(trainer_state['log_history'])
+
+plt.plot(loss_history['step'], loss_history['loss'], label='Training Loss')
+plt.title("Loss History")
+plt.xlabel("Epochs")
+plt.ylabel("Loss")
+plt.show()
+
+inference_tracker = EmissionsTracker()
+inference_tracker.start()
```
Comment (umoqnier) [inline practicas/David/p6/practica_6.py:131]: Nice :leafy_green:

File: practicas/David/p6/Dataset_ticket.xlsx
Comment (umoqnier) [inline practicas/David/p6/Dataset_ticket.xlsx:1]: Creo que esto no debería estar aquí. Borralo para el futuro

File: practicas/David/p6/Readme.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+Para instalar las bibliotecas necesarias, puede utilizar el siguiente comando:
+```bash
+pip install datasets transformers evaluate pandas torch codecarbon numpy
+```
+
+## Ejecución
+Para ejecutar los scripts de la práctica, asegúrese de tener todas las bibliotecas instaladas y ejecute los archivos correspondientes desde su entorno de desarrollo.
+
+## Demo
+Puede acceder a la demo del modelo en Hugging Face Spaces [aquí](https://huggingface.co/spaces/davidpmijan/movie-genres).
```
Comment (umoqnier) [inline practicas/David/p6/Readme.md:60]: Como esto es de sumo interés para esta práctica, conviene que esté más a la vista, quizá ponerlo inmediatamente después de la descripción inicial, no hasta el fondo

**Review by umoqnier — APPROVED** [review]
# Calificaciones

Se califican las prácticas 5 y 6, ya que se incluyó el código de ambas en el PR

## Práctica 5: 9 + EXTRA
**Penalización por entrega tardía**
- En general se ve bien la práctica, aunque un poco *messy* por todo el código que está de la sesión
- Veo que hiciste el extra efectivamente, bien :)

## Práctica 6: 8 + EXTRA
**Penalización por entrega tardía**
- Veo la documentación bastante bien y el código bien estructurado.
- Tambien veo que agregaste el extra midiendo las emisiones
- Lamentablemente no veo las reflexiones finales solicitadas
> - Reporta que tan bien se resolvió la tarea y que tan útil fue tu app
> - Reporta retos y dificultades al realizar el fine-tuning y al poner tu modelo en producción
